src/unstract/sdk/utils/common_utils.py

```diff
@@ -54,3 +55,35 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator
+
+
+def capture_metrics(func):
+    """Decorator to capture metrics at the start and end of a function."""
+
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        metrics_mixin = None
+        TIME_TAKEN_KEY = "time_taken(s)"
```

Comment:
NIT: Constant can be moved to `MetricsMixin` class or module and referred from there. Will be a single place to change / update this